src/compressor/lz4/CMakeLists.txt

```diff
@@ -5,7 +5,10 @@ set(lz4_sources
 )
 
 add_library(ceph_lz4 SHARED ${lz4_sources})
-target_link_libraries(ceph_lz4 PRIVATE LZ4::LZ4)
+target_link_libraries(ceph_lz4 LZ4::LZ4)
+if(HAVE_QATZIP)
```

Comment:
instead of linking all supported compressors against qatzip, can we just `add_library(compressor ...)` and link compressor to it instead? as it's compressor which pulls `QatAccel.cc` in. and it's `QatAccel.cc` which uses qatzip.

ceph_lz4 and friends just use this feature offered by compressor.

Reply:
There is "add_library(cephd_compressor_base STATIC ${compressor_srcs})" in src/compressor/CMakeLists.txt of previous version, and I can link libqatzip.so to cephd_compressor_base. But I don't know why it was removed and so no available object can be linked by qatzip library. I have to change CMake file of specific plugin to link this.

Reply:
> There is "add_library(cephd_compressor_base STATIC ${compressor_srcs})" in src/compressor/CMakeLists.txt of previous version

could you point me to the sha1 of that version?

Reply:
this is the PR for initial implementation, you can check it: https://github.com/ceph/ceph/pull/19714/

Reply:
they you were referencing the case where we tried to avoid linking dynamic libraries. it was for a dedicated build only for rook where `WITH_EMBEDDED` is ON.

Reply:
> I have to change CMake file of specific plugin to link this.

i still think:

> instead of linking all supported compressors against qatzip, can we just `add_library(compressor ...)` and link compressor to it instead? as it's compressor which pulls `QatAccel.cc` in. and it's `QatAccel.cc` which uses qatzip.

unless we have no better options.

Reply:
Yes, if so, maybe I have to recover to previous CMake file version which can work, to link libqatzip library to cephd_compressor_base.